src/main/java/org/wildfly/security/sasl/digest/AbstractDigestMechanism.java

```diff
@@ -439,46 +415,36 @@ protected DigestWrapper(boolean confidential) {
 
         byte[] cipheredPart = null;
         try {
-            cipheredPart = wrapCipher.doFinal(toCipher);
+            cipheredPart = wrapCipher.update(toCipher);
         } catch (Exception e) {
             throw new SaslException("Problem during crypt.", e);
         }
 
         byte[] result = new byte[cipheredPart.length + 6];
         System.arraycopy(cipheredPart, 0, result, 0, cipheredPart.length);
-        integerByteOrdered(1, result, cipheredPart.length, 2);  // 2-byte message type number in network byte order with value 1
-        integerByteOrdered(seqNum, result, cipheredPart.length + 2, 4); // 4-byte sequence number in network byte order
+        DigestUtils.integerByteOrdered(1, result, cipheredPart.length, 2);  // 2-byte message type number in network byte order with value 1
+        DigestUtils.integerByteOrdered(wrapSeqNum, result, cipheredPart.length + 2, 4); // 4-byte sequence number in network byte order
 
-        seqNum++;
+        wrapSeqNum++;
         return result;
     }
 
-    private byte[] computeHMAC(byte[] kc, int sequenceNumber, byte[] message, int offset, int len) throws SaslException {
-        Mac mac = getHmac();
-        SecretKeySpec ks = new SecretKeySpec(kc, HMAC_algorithm);
-        try {
-            mac.init(ks);
-        } catch (InvalidKeyException e) {
-            throw new SaslException("Invalid key provided", e);
-        }
-        byte[] buffer = new byte[len + 4];
-        integerByteOrdered(sequenceNumber, buffer, 0, 4);
-        System.arraycopy(message, offset, buffer, 4, len);
-        return mac.doFinal(buffer);
-    }
-
     private byte[] unwrapConfidentialityProtectedMessage(byte[] message, int offset, int len) throws SaslException {
 
-        int messageType = decodeByteOrderedInteger(message, offset + len - 6, 2);
-        int extractedSeqNum = decodeByteOrderedInteger(message, offset + len - 4, 4);
+        int messageType = DigestUtils.decodeByteOrderedInteger(message, offset + len - 6, 2); // type and seqnum is 6 last bytes
+        int extractedSeqNum = DigestUtils.decodeByteOrderedInteger(message, offset + len - 4, 4);
```

Comment:
Not a big deal but probably could have used static imports for these method calls to the DigestUtils doesn't need to be referenced in so many places.